castai/resource_service_account.go

```diff
@@ -0,0 +1,570 @@
+package castai
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/hashicorp/go-cty/cty"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+
+	"github.com/castai/terraform-provider-castai/castai/sdk"
+)
+
+const (
+	FieldServiceAccountOrganizationID = "organization_id"
+	FieldServiceAccountName           = "name"
+	FieldServiceAccountID             = "service_account_id"
+	FieldServiceAccountDescription    = "description"
+	FieldServiceAccountEmail          = "email"
+
+	FieldServiceAccountAuthor      = "author"
+	FieldServiceAccountAuthorID    = "id"
+	FieldServiceAccountAuthorEmail = "email"
+	FieldServiceAccountAuthorKind  = "kind"
+
+	FieldServiceAccountKeyID               = "id"
+	FieldServiceAccountKeyOrganizationID   = "organization_id"
+	FieldServiceAccountKeyServiceAccountID = "service_account_id"
+	FieldServiceAccountKeyName             = "name"
+	FieldServiceAccountKeyPrefix           = "prefix"
+	FieldServiceAccountKeyLastUsedAt       = "last_used_at"
+	FieldServiceAccountKeyExpiresAt        = "expires_at"
+	FieldServiceAccountKeyActive           = "active"
+)
+
+func resourceServiceAccount() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceServiceAccountCreate,
+		ReadContext:   resourceServiceAccountRead,
+		UpdateContext: resourceServiceAccountUpdate,
+		DeleteContext: resourceServiceAccountDelete,
+
+		Description: "Service account resource allows managing CAST AI service accounts.",
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(3 * time.Minute),
+			Read:   schema.DefaultTimeout(3 * time.Minute),
+			Update: schema.DefaultTimeout(3 * time.Minute),
+			Delete: schema.DefaultTimeout(3 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			FieldServiceAccountOrganizationID: {
+				Type:             schema.TypeString,
+				Required:         true,
+				ForceNew:         true,
+				Description:      "ID of the organization.",
+				ValidateDiagFunc: validation.ToDiagFunc(validation.IsUUID),
+			},
+			FieldServiceAccountName: {
+				Type:             schema.TypeString,
+				Required:         true,
+				Description:      "Name of the service account.",
+				ValidateDiagFunc: validation.ToDiagFunc(validation.StringIsNotWhiteSpace),
+			},
+			FieldServiceAccountDescription: {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "Description of the service account.",
+			},
+			FieldServiceAccountEmail: {
```

Comment:
Do we also need to provide the service account id field?

Reply:
Could you elaborate on that? I'm not sure what you meant